gulpfile.js

```diff
@@ -129,26 +146,35 @@ gulp.task('build-ci', ['clean'], function (cb) {
 });
 
 gulp.task('build-test', function buildTest(cb) {
-    var config = _.defaultsDeep({}, require('./webpack.config'));
-
-    config.entry = {};
-    // the key in the entry, formed by path.join, determines the target folder structure of the test file
-    config.entry[path.join('request', 'repository', 'request-repository-cache')] = './test/request/repository/request-repository-cache.spec.ts';
-
-    config.output.path = testSettings.output;
-
-    var started = false;
-    function processResult(err, stats) {
-        gutil.log('Webpack\n' + stats.toString(config.log));
-
-        if (!started) {
-            started = true;
-            cb();
+    getTests(function collectAllTests(tests) {
+        var config = _.defaultsDeep({}, require('./webpack.config'));
+        var pathPrefix = path.join(testSettings.input, '/');
+        var re = new RegExp(testSettings.testSuffix + '$');
+
+        config.entry = {};
+        tests.forEach(function collectTest(test) {
+            var pathName = test.replace(pathPrefix, '');
```

Comment:
Here we want to strip out the prefix from the full path name. Replacing the prefix with `''` should be fine. I added some comments to make the intent more clear.